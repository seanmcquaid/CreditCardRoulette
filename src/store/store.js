import React, {createContext, useReducer} from "react";
import rootReducer from "../reducers/rootReducer";

const initialState = {
    checkAmount : 0,
    playerNames : []
};

export const GlobalContext = createContext();

export const GlobalContextProvider = props => {
    const [state, dispatch] = useReducer(rootReducer, initialState);

    return (
        <GlobalContext.Provider value={[state, dispatch]}>
            {props.children}
        </GlobalContext.Provider>
    )
};
