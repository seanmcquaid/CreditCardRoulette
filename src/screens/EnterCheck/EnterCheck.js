import React, {useState, useContext} from "react";
import {GlobalContext} from "../../store/store";
import {View} from "react-native";
import {setCheckAmount} from "../../actions/actions";
import styled from "styled-components/native";
import HeaderText from "../../components/HeaderText/HeaderText";
import Button from "../../components/Button/Button";
import TextInput from "../../components/TextInput/TextInput";

const EnterCheck = ({navigation}) => {
    const [state, dispatch] = useContext(GlobalContext);
    const [inputText, setInputText] = useState("");
    
    const handlePlayButtonTap = async () => {
        await dispatch(setCheckAmount(inputText));
        await navigation.navigate("Players List");
    };

    return (
        <EnterCheckScreenContainer>
            <HeaderText>Enter Check</HeaderText>
            <TextInput 
                placeholder="Enter Amount Here" 
                value={inputText} 
                onChangeText={text => setInputText(text)}
            />
            <Button title="Play" onPress={() => handlePlayButtonTap()}/>
        </EnterCheckScreenContainer>
    );
};

const EnterCheckScreenContainer = styled(View)`
    flex : 1;
    justify-content : center;
    align-items : center;
    background-color : #00ae6b;
`;


export default EnterCheck;