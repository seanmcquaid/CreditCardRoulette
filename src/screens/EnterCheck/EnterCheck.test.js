import React from "react";
import EnterCheck from "./EnterCheck";
import {render, fireEvent, act, cleanup} from "react-native-testing-library";
import {GlobalContextProvider} from "../../store/store";

describe("<EnterCheck/>", () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    })
    it("Text Input works properly", () => {
        const navigate = jest.fn();

        const props = {
            navigation : {
                navigate
            }
        };
        
        const {getByTestId} = render(
        <GlobalContextProvider>
            <EnterCheck {...props}/>
        </GlobalContextProvider>);


        fireEvent.changeText(getByTestId("textInput"), "Hello");
        expect(getByTestId("textInput").props.value).toEqual("Hello");
    });

    it("handlePlayButtonOnTapHandler with just numbers", async () => {
        const navigate = jest.fn();
        
        const props = {
            navigation : {
                navigate
            }
        };
        
        const {getByTestId} = render(
        <GlobalContextProvider>
            <EnterCheck {...props}/>
        </GlobalContextProvider>);

        fireEvent.changeText(getByTestId("textInput"), "366");
        await fireEvent.press(getByTestId("playButton"));

        expect(navigate).toHaveBeenCalledTimes(1);
        expect(navigate).toHaveBeenCalledWith("Players List");
    });

    it("handlePlayButtonOnTapHandler with no numbers numbers", () => {
        const navigate = jest.fn();
        
        const props = {
            navigation : {
                navigate
            }
        };
        
        const {getByTestId} = render(
        <GlobalContextProvider>
            <EnterCheck {...props}/>
        </GlobalContextProvider>);

        fireEvent.changeText(getByTestId("textInput"), "SEAN");
        fireEvent.press(getByTestId("playButton"));

        expect(navigate).toHaveBeenCalledTimes(0);
    });
    it("Matches snapshot correctly", () => {
        const navigate = jest.fn();
        
        const props = {
            navigation : {
                navigate
            }
        };
        
        const enterCheck = render(
        <GlobalContextProvider>
            <EnterCheck {...props}/>
        </GlobalContextProvider>);

        expect(enterCheck).toMatchSnapshot();
    });
});