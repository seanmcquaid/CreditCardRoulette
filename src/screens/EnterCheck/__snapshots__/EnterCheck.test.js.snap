// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnterCheck/> Matches snapshot correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#00ae6b",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Fira-Sans",
          "fontSize": 26,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    Enter Check
  </Text>
  <View
    style={
      Array [
        Object {
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChange={[Function]}
      placeholder="Enter Amount Here"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderStyle": "solid",
            "borderWidth": 2,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 180,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C5B358",
          "borderColor": "#000",
          "borderRadius": 3,
          "borderStyle": "solid",
          "borderWidth": 2,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 16,
          "paddingBottom": 16,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 16,
        },
      ]
    }
    testID="button"
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Merriweather-Regular",
            "fontSize": 16,
          },
          null,
        ]
      }
      testID="buttonTitle"
    >
      Play
    </Text>
  </View>
</View>
`;
